packages/elasticsearch-client/src/config/default/resources/index-document.ts

```diff
@@ -16,40 +16,50 @@
 
 import { ReferrableResource } from '@tech-matters/types';
 
-export type ResourcesIndexDocument = {
-  name: string;
-  text1: string[];
-  text2: string[];
-};
-
-// This is a list of attribute names that should be given higher priority in search results.
-const HIGH_PRIORITY_ATTRIBUTES = ['title'];
+import {
+  isHighBoostGlobalField,
+  isMappingField,
+  mappingFields,
+  ResourcesIndexDocument,
+} from './config';
 
+// TODO: when we have more than one index and config type, we should probably make this a little more generic
+// and just import the config to generate it. Leaving here for now.
 export const convertDocument = (resource: ReferrableResource): ResourcesIndexDocument => {
   const { name } = resource;
```

Comment:
In our data structure, name is a single field, but the example payload from the client has multiple languages. Will name exist both as a property of the top level resource and as a regular attribute property that may have multiple languages?

If so, this should probably be refactored to use the normal mapFields setup.

Reply:
I imagine there will be both in a lot of cases. The name on the DB record is intended to be a universal, human readable handle that we can assume exists across any resource DB implementation, but other, helpline specific 'name' type attributes might well exist in the helpline's specific schema - there could be several of these, depending on the helpline's requirements (like KHP have a 'name' and and 'official name'), and they could be multilingual

Reply:
Excellent. I'll leave as is for now and we can address the final tweaks to the mapping in one of the future ticket.